tests/ops/qubit/test_qchem_ops.py

```diff
@@ -44,32 +44,34 @@ def test_single_excitation_plus_decomp(self, phi):
         and the target is wire 0 in the decomposition. (Not applicable for
         ControlledPhase as it has the same matrix representation regardless of the
         control and target wires.)"""
-        decomp = qml.SingleExcitationPlus(phi, wires=[0, 1]).decompose()
-
-        mats = []
-        for i in reversed(decomp):
-            if i.wires.tolist() == [0]:
-                mats.append(np.kron(i.matrix, np.eye(2)))
-            elif i.wires.tolist() == [1]:
-                mats.append(np.kron(np.eye(2), i.matrix))
-            elif i.wires.tolist() == [1, 0] and isinstance(i, qml.CRY):
-                new_mat = np.array(
-                    [
-                        [1, 0, 0, 0],
-                        [0, np.cos(phi / 2), 0, -np.sin(phi / 2)],
-                        [0, 0, 1, 0],
-                        [0, np.sin(phi / 2), 0, np.cos(phi / 2)],
-                    ]
-                )
-
-                mats.append(new_mat)
-            else:
-                mats.append(i.matrix)
-
-        decomposed_matrix = np.linalg.multi_dot(mats)
-        exp = SingleExcitationPlus(phi)
-
-        assert np.allclose(decomposed_matrix, exp)
+        decomp1 = qml.SingleExcitationPlus(phi, wires=[0, 1]).decomposition()
+        decomp2 = qml.SingleExcitationPlus.compute_decomposition(phi, wires=[0, 1])
+
+        for decomp in [decomp1, decomp2]:
+            mats = []
+            for i in reversed(decomp):
+                if i.wires.tolist() == [0]:
+                    mats.append(np.kron(i.matrix, np.eye(2)))
+                elif i.wires.tolist() == [1]:
+                    mats.append(np.kron(np.eye(2), i.matrix))
+                elif i.wires.tolist() == [1, 0] and isinstance(i, qml.CRY):
+                    new_mat = np.array(
+                        [
+                            [1, 0, 0, 0],
+                            [0, np.cos(phi / 2), 0, -np.sin(phi / 2)],
+                            [0, 0, 1, 0],
+                            [0, np.sin(phi / 2), 0, np.cos(phi / 2)],
+                        ]
+                    )
+
+                    mats.append(new_mat)
+                else:
+                    mats.append(i.matrix)
```

Comment:
maybe this can be replaced now with a call to `qml.transforms.get_unitary_matrix`?

Reply:
Honestly didn't look too much in the code; just wrapped it in a for loop over both decomposition methods.  A lot of this testing code would be substantially cleaner if we can a way to assert operation equality.